include/Frame.h

```diff
@@ -156,7 +156,7 @@ namespace openshot
 		void AddColor(int new_width, int new_height, std::string new_color);
 
 		/// Add (or replace) pixel data to the frame
-		void AddImage(int new_width, int new_height, int bytes_per_pixel, QImage::Format type, const unsigned char *pixels_);
+		void AddImage(int new_width, int new_height, int bytes_per_pixel, int bytes_per_line, QImage::Format type, const unsigned char *pixels_);
```

Comment:
Perhaps it'd be better to add this as an additional overload, rather than redefining the existing `Frame::AddImage()` semantics? There may be external API consumers who rely on the previous signature.

To make overload resolution slightly less harrowing, there's _some_ argument for tacking `bytes_per_line` on at the very end. Arguably it's a detail specific to the **buffer**, not a parameter describing the image (like all of the other parameters surrounding it), so it should (optionally) follow the buffer it describes — and only in the cases where it's necessary to specify it. (IOW, only when `bytes_per_line != width * bytes_per_pixel`.)

Reply:
A lot of good questions there, and I'm open to any solution. Here's my thoughts:

- I don't consider the Frame class part of the externally exposed API of libopenshot, so since there are no other uses of this function, I felt safe changing it. If we consider the Frame class as something external callers can use, then I agree it should be a separate overload.

- I see your point about bytes_per_line at the end; it doesn't matter to me, and I see the argument that it's a "property" of buffer.